Common/Data/Format/IniFile.cpp

```diff
@@ -159,31 +177,40 @@ static std::string EscapeComments(const std::string &value) {
 	return result;
 }
 
+void ParsedIniLine::ParseFrom(const std::string &line) {
+	// TODO: Skip whitespace.
+	if (line[0] == '#') {
+		key.clear();
+		value.clear();
+		comment = line;
+	} else {
+		ParseLine(line, &key, &value, &comment);
+	}
+}
+
+void ParsedIniLine::Reconstruct(std::string *output) const {
+	if (!key.empty()) {
+		*output = EscapeHash(key) + " = " + EscapeHash(value) + comment;
+	} else {
+		*output = comment;
+	}
+}
+
 void Section::Clear() {
-	lines.clear();
+	lines_.clear();
 }
 
-std::string* Section::GetLine(const char* key, std::string* valueOut, std::string* commentOut)
-{
-	for (std::vector<std::string>::iterator iter = lines.begin(); iter != lines.end(); ++iter)
-	{
-		std::string& line = *iter;
-		std::string lineKey;
-		ParseLine(line, &lineKey, valueOut, commentOut);
-		if (!strcasecmp(lineKey.c_str(), key))
+ParsedIniLine *Section::GetLine(const char *key) {
+	for (auto &line : lines_) {
+		if (!strvcasecmp(line.Key(), key))
```

Comment:
How often do we call this?  Probably a lot?  Maybe a map would be worth it... though it'd have to be with a lowercased key or something.

-[Unknown]

Reply:
Not enough to be worth it I think, unless we start using truly massive ini files - this is already a huge, huge speedboost over parsing and allocating over and over.